components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -1710,6 +1715,21 @@ private String getSpTenantDomain(String clientId) throws InvalidRequestException
         }
     }
 
+    private String getLoginTenantDomain(OAuthMessage oAuthMessage, String clientId) throws InvalidRequestException {
+
+        if (!IdentityTenantUtil.isTenantedSessionsEnabled()) {
+            return getSpTenantDomain(clientId);
```

Comment:
If the "tenanted session ... " configuration is disabled we set the path of the cookie as '/' right? In that case why we return the SP tenant domain in this case? IMO regardless of the above configuration, we should return users' tenant domain.

Reply:
Yes. But If the "tenanted session ... " configuration is disabled we can't get the users' tenant domain, since we get it from the oauth request, 't parameter'. So in this case we return the SP tenant domain.